ceph/ceph_admin/bootstrap.py

```diff
@@ -52,22 +55,124 @@ def construct_registry(cls, registry: str, json_file: bool = False):
     return config_dict_to_string(reg_args)
 
 
-class BootstrapMixin:
-    """Add bootstrap support to the child class."""
+def validate_fsid(cls, fsid: str, out: str) -> bool:
+    """
+    Method is used to validate fsid
+    Args:
+        cls: class object
+        fsid: <ID>
+        out: bootstrap response logs
 
-    def get_method(self, key):
-        """
-        Fetch method using the Key
-        Args:
-            key: fsid
+    Returns: Boolean value whether fsid is valid
+
+    """
+    out, err = cls.shell(args=["ceph", "fsid"])
+    return out == fsid
 
-        Returns: Returns the method based on the Key
 
-        """
-        verify_map = {
-            "fsid": self.validate_fsid,
+def validate_dashboard(cls, out, user=None, password=None):
+    """
+    Method to validate dashboard login using provided user/password
+    and Also validates user/passwd in bootstrap console output
+
+    Args:
+        cls: class object
+        out: bootstrap response log
+        user: dashboard user
+        password: dashbard password
+
+    Returns:
+        boolean
+    """
+    _, _, db_log_part = out.partition("Ceph Dashboard is now available at:")
+
+    if not db_log_part:
+        return False
+
+    def parse(string):
+        str_ = r"%s:\s(.*)" % string
+        return re.search(str_, db_log_part).group(1)
+
+    host = parse("URL")
+    user_ = parse("User")
+    passwd_ = parse("Password")
+
+    if user:
+        assert user == user_
+    if password:
+        assert password == passwd_
+
+    host_ = re.search(r"https?:[/]{2}(.*):8443", host).group(1)
+    for node in cls.cluster.get_nodes():
+        if host_ in node.shortname:
+            host = host.replace(host_, node.ip_address)
+            break
+    else:
+        return False
+
+    data = json.dumps({"password": passwd_, "username": user_})
+    url_ = f"{host}api/auth"
+
+    session = requests.Session()
+    session.headers = {}
```

Comment:
Can we collapse the assignment and `.update()` call here?

```suggestion
    session.headers = {
            "accept": "application/vnd.ceph.api.v1.0+json",
            "content-type": "application/json",
        }
```

Reply:
done